packages/cadl-ranch-specs/http/models/property-types/main.tsp

```diff
@@ -183,3 +183,50 @@ model NeverProperty is ModelTemplate<never>;
 @route("/never")
 @operationGroup
 interface Never extends ModelOperations<NeverProperty, "<don't include this property>"> {}
+
+// Test a model with a model property which could be nullable
+@doc("Model with nullable properties")
+model ModelWithNullableProperty {
+  @doc("id property")
+  id: string;
+
+  @doc("nullable property")
+  property: string | null;
+}
+
+@doc("Model with a property null. The null value will be set in application/merge-patch+json case.")
+@route("/nullable-property")
+@operationGroup
+interface ModelNullableProperty {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing"
+  @scenario
+  @scenarioDoc("""
+Expected response body::
+  ```json
+  {
+    "id": 1,
+    "property": "foo"
+  }
+  ```
+""")
+  @get
+  @doc("Get call")
+  get(): ModelWithNullableProperty;
```

Comment:
Should we maybe not test the case where the service returns `null` instead(Or maybe both)

Reply:
Similar optional property spec I keep both the getAll and getDefault. The first would return non-null value the latter would be a null.

Reply:
awesome!